### PR TITLE
WT-12324 Prefetch locking

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/btree/bt_prefetch.c [hunk 1]
```diff
@@ -25,11 +25,7 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
     conn = S2C(session);
     block_preload = 0;
 
-    /*
-     * Do not add refs from trees that have eviction disabled since they are probably being closed.
-     */
-    if (!(F_ISSET(ref, WT_REF_FLAG_LEAF)) || S2BT(session)->evict_disabled > 0 ||
-      (__wt_session_gen(session, WT_GEN_SPLIT) == 0)) {
+    if (!(F_ISSET(ref, WT_REF_FLAG_LEAF) || __wt_session_gen(session, WT_GEN_SPLIT) == 0)) {
```
Comment (luke-pearson) [inline src/btree/bt_prefetch.c:28]: I'm not sure why we check the session gen here, if anyone knows please clarify.

File: src/conn/conn_prefetch.c [hunk 2]
```diff
@@ -98,21 +98,24 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
...
-        if (pe == NULL)
+        if (pe == NULL) {
+            __wt_spin_unlock(session, &conn->prefetch_lock);
             break;
+        }
 
         TAILQ_REMOVE(&conn->pfqh, pe, q);
         --conn->prefetch_queue_count;
 
+        ((WT_BTREE *)pe->dhandle->handle)->prefetch_busy += 1;
```
Comment (luke-pearson) [inline src/conn/conn_prefetch.c:112]: If this is only ever 1 or 0 we should convert it to a boolean.

File: src/conn/conn_prefetch.c [hunk 3]
```diff
@@ -163,6 +166,12 @@ __wt_conn_prefetch_queue_push(WT_SESSION_IMPL *session, WT_REF *ref)
     pe->dhandle = session->dhandle;
 
     __wt_spin_lock(session, &conn->prefetch_lock);
+    /* Don't queue pages for trees that have eviction disabled. */
+    if (S2BT(session)->evict_disabled > 0) {
```
Comment (luke-pearson) [inline src/conn/conn_prefetch.c:170]: I moved the evict_disabled check inside of the lock which is the main thing.

File: src/conn/conn_prefetch.c (hunk 2 above)
Comment (kommiharibabu) [reply src/conn/conn_prefetch.c:112]: Without an atomic increment, this value can go backward. A similar problem was observed with `btree->write_gen` and later converted into an atomic increment.

Comment (luke-pearson) [reply src/conn/conn_prefetch.c:112]: Okay so I'll convert this back to an atomic load and store.

**Comment by luke-pearson** [comment]
Closing as i have migrated these changes to #10189

**Pull request closed without merging**